Sources/FirebaseFirestore/Firestore+Swift.swift

```diff
@@ -44,6 +44,11 @@ public class Firestore {
     }
   }
 
+  public func clearPersistence(completion: @escaping () -> Void) {
+    let future = swift_firebase.swift_cxx_shims.firebase.firestore.firestore_clear_persistence(self)
+    future.setCompletion(completion)
+  }
```

Comment:
A couple things we should change about this for consistency with the Obj C API. We should actually provide two versions of this function, a completion callback version like you have done here and an `async` function version. The completion callback version should run the completion callback on the main thread. This is all to mimic the behavior of the Obj C API.

See examples in DocumentReference+Swift.swift and Query+Swift.swift

```
public func clearPersistence(completion: @escaping () -> Void) {
  let future = swift_firebase.swift_cxx_shims.firebase.firestore.firestore_clear_persistence(self)
  future.setCompletion({
    DispatchQueue.main.async {
      completion()
    }
  })
}

public func clearPersistence() async {
  withCheckedContinuation { continuation in
    let future = swift_firebase.swift_cxx_shims.firebase.firestore.firestore_clear_persistence(self)
    future.setCompletion({
      continuation.resume()
    }
  }
}
```

Reply:
ah ok - thank you!
I also see some compilation issues for some reason :\ looking